src/toil/batchSystems/abstractGridEngineBatchSystem.py

```diff
@@ -116,7 +117,22 @@ def createJobs(self, newJob):
                 logger.debug("Running %r", subLine)
 
                 # submit job and get batch system ID
-                batchJobID = self.submitJob(subLine)
+                success = False
+                retries = 3
+                latest_err = None
+                while not success and retries:
+                    try:
+                        batchJobID = self.submitJob(subLine)
+                        success = True
+                    except subprocess.CalledProcessError as err:
+                        latest_err = err
+                        logger.error(
+                            "submitJob failed with code %d: %s",
+                            err.returncode, err.output)
+                    finally:
+                        retries -= 1
+                if latest_err:
```

Comment:
I think you may want to add `and not success` here.  Otherwise, if you have a failure, then retry and it succeeds, it looks like it will still have the error saved and raise.